Added documentation for Ignored Build Step (#2117)
vercel/docs: pages/docs/platform/projects.mdx

  Selecting the <b>Security</b> menu item from the Project Settings page.
</Caption>

### Ignored Build Step

> By default, <ProductName /> will issue a new Deployment with Build (unless the Build Step is [skipped](/docs/build-step#build-command)) for every commit to your connected Git repository.

Reviewer: ```suggestion
By default, <ProductName /> will issue a new [Deployment](/docs/platform/deployments) with Build (unless the Build Step is [skipped](/docs/build-step#build-command)) for every commit to your connected Git repository.
```

  Selecting the <b>Security</b> menu item from the Project Settings page.
</Caption>

### Ignored Build Step

By default, <ProductName /> will issue a new Deployment with Build (unless the Build Step is [skipped](/docs/build-step#build-command)) for every commit to your connected Git repository.

If the SHA of the commit was already deployed in the past, however, no new Deployment is created. In that case, the last Deployment matching that SHA is returned instead.

> You can learn more about this behavior [here](/docs/platform/deployments?query=dedup#deduplication).

Reviewer 2: ```suggestion
You can learn more about this behavior in the [Deduplication](/docs/platform/deployments#deduplication) section.
```

Author: This text doesn't exist anymore.

To customize the behavior for your Git repositories, you can enter a command in the **Ignored Build Step** section of the **Git Integration** page, which will then determine whether or not a new Build needs to be issued:

<Image
  src={`${process.env.ASSETS}/docs/platform/projects/ignored-build-step.png`}
  width={1488 / 2}
  height={616 / 2}
  shadow
/>
<Caption>Specify the Git repository for your project.</Caption>

If the SHA of your commit wasn't deployed before, a regular Deployment is issued in the `QUEUED` state, which then moves into the `BUILDING` state once it starts building.

> While the Deployment is in the `BUILDING` state, the command you've entered in the **Ignored Build Step** section will be run. If the command exits with code `1`, the Build will continue as usual. If the command exits with code `0`, however, the Build is immediately aborted and the Deployment state is set to `CANCELED`.

Reviewer 2: ```suggestion
While the Deployment is in the `BUILDING` state, the command you've entered in the **Ignored Build Step** section will be run. If the command exits with code `0`, the Build will continue as usual. If the command exits with code `1`, the Build is immediately aborted and the Deployment state is set to `CANCELED`.
```

I think you have these mixed up, exit code 0 is success

Author: Nope. It's intentional.
